dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/DependencyConflict.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.opensource.classpath;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.cloud.tools.opensource.dependencies.Artifacts;
+import com.google.cloud.tools.opensource.dependencies.DependencyPath;
+import java.util.Objects;
+import org.eclipse.aether.artifact.Artifact;
+
+/**
+ * Diamond dependency conflict caused a {@link LinkageProblem} where the {@link LinkageProblem}'s
+ * invalid reference points to the symbol in {@code pathToSelectedArtifact.getLeaf()} but a valid
+ * symbol is in {@code pathToUnselectedArtifact.getLeaf()}.
+ */
+class DependencyConflict extends LinkageProblemCause {
+  private Symbol symbol;
+  private DependencyPath pathToUnselectedArtifact;
+  private DependencyPath pathToSelectedArtifact;
+
+  DependencyConflict(
+      Symbol symbol,
+      DependencyPath pathToSelectedArtifact,
+      DependencyPath pathToUnselectedArtifact) {
+    this.symbol = checkNotNull(symbol);
+    this.pathToUnselectedArtifact = checkNotNull(pathToUnselectedArtifact);
+    this.pathToSelectedArtifact = checkNotNull(pathToSelectedArtifact);
+  }
+
+  public DependencyPath getPathToUnselectedArtifact() {
```

Comment:
How does this handle the case where there's more than one unselected artifact? I.e. when there's three or more versions in the tree?

Reply:
The first artifact is enough. This method tells which artifact was used when the problematic artifact (that has dependency conflicts) was built.

Reply:
pathToTheArtifactThruSource.

path to the artifact which was used when building the artifact that source class belongs to.